source/write/gridfs.txt

```diff
@@ -0,0 +1,222 @@
+operation is performed. The driver creates indexes only if they don't exist and when the
+bucket is empty. For more information about
+GridFS indexes, see :manual:`GridFS Indexes </core/gridfs/#gridfs-indexes>`
+in the {+mdb-server+} manual.
+
+When storing files with GridFS, the driver splits the files into smaller
+chunks, each represented by a separate document in the ``chunks`` collection.
+It also creates a document in the ``files`` collection that contains
+a file ID, file name, and other file metadata. You can upload the file from
+memory or from a stream. See the following diagram to see how GridFS splits
+the files when uploaded to a bucket.
+
+.. figure:: /includes/figures/GridFS-upload.png
+   :alt: A diagram that shows how GridFS uploads a file to a bucket
+
+When retrieving files, GridFS fetches the metadata from the ``files``
+collection in the specified bucket and uses the information to reconstruct
+the file from documents in the ``chunks`` collection. You can read the file
+into memory or output it to a stream.
+
+Create a GridFS Bucket
+----------------------
+
+To store or retrieve files from GridFS, create a GridFS bucket by calling the
+``GridFSBucket()`` constructor and passing in a ``MongoDatabase`` instance.
+You can use the ``GridFSBucket`` instance to
+call read and write operations on the files in your bucket.
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-create-bucket
+      :end-before: end-create-bucket
+
+To create or reference a bucket with a custom name other than the default name
+``fs``, pass your bucket name as the second parameter to the ``GridFSBucket()``
+constructor, as shown in the following example:
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-create-custom-bucket
+      :end-before: end-create-custom-bucket
+
+Upload Files
+------------
+
+The ``GridFSBucket.uploadFromObservable()`` method reads the contents of an
+``Observable[ByteBuffer]`` and saves it to the ``GridFSBucket`` instance.
+
+You can use the ``GridFSUploadOptions`` type to configure the chunk size or include
+additional metadata.
+
+The following example uploads the contents of an ``Observable[ByteBuffer]`` into
+``GridFSBucket``:
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-upload-files
+      :end-before: end-upload-files
+
+Retrieve File Information
+-------------------------
+
+In this section, you can learn how to retrieve file metadata stored in the
+``files`` collection of the GridFS bucket. The metadata contains information
+about the file it refers to, including:
+
+- The ``_id`` of the file
+- The name of the file
+- The length/size of the file
+- The upload date and time
+- A ``metadata`` document in which you can store any other information
```

Comment:
Q: Is there a link to show what these metadata fields are named? Think it would be helpful to include.